pkg/slayers/scion_spec.gobra

```diff
@@ -364,7 +363,14 @@ func (s *SCION) EQAbsHeader(ub []byte) bool {
 		// Might be worth introducing EQAbsHeader as an interface method on Path
 		// to avoid doing these casts, especially when we add support for EPIC.
 		typeOf(s.Path) == (*scion.Raw)                               &&
-		s.Path.(*scion.Raw).EQAbsHeader(ub[CmnHdrLen+s.AddrHdrLenSpecInternal() : s.HdrLen*LineLen])
+		unfolding acc(s.Path.Mem(ub[CmnHdrLen+s.AddrHdrLenSpecInternal() : s.HdrLen*LineLen]), _) in
+		unfolding acc(slices.AbsSlice_Bytes(ub, 0, len(ub)), _) in
+		let _ := Asserting(forall k int :: {&ub[CmnHdrLen+s.AddrHdrLenSpecInternal() : s.HdrLen*LineLen][k]} 0 <= k && k < s.HdrLen*LineLen ==>
+			&ub[CmnHdrLen+s.AddrHdrLenSpecInternal() : s.HdrLen*LineLen][k] == &ub[CmnHdrLen+s.AddrHdrLenSpecInternal() + k]) in
+		let _ := Asserting(forall k int :: {&ub[CmnHdrLen+s.AddrHdrLenSpecInternal() : s.HdrLen*LineLen][:scion.MetaLen][k]} 0 <= k && k < scion.MetaLen ==>
+			&ub[CmnHdrLen+s.AddrHdrLenSpecInternal() : s.HdrLen*LineLen][:scion.MetaLen][k] == &ub[CmnHdrLen+s.AddrHdrLenSpecInternal() : s.HdrLen*LineLen][k]) in
+		s.Path.(*scion.Raw).Base.GetMetaHdr() ==
+		scion.DecodedFrom(binary.BigEndian.Uint32(ub[CmnHdrLen+s.AddrHdrLenSpecInternal() : s.HdrLen*LineLen][:scion.MetaLen]))
```

Comment:
this is a bit hard to read. There are many repeated expressions that we should maybe store in variables. As a sugesstion, we should have at least a `low := CmnHdrLen+s.AddrHdrLenSpecInternal()`, `high := s.HdrLen*LineLen`, `ubPath := ub[low:high]`